native-assets/fractional-NFT/project/contracts/f-NFT-contract/tests/functions/deposit.rs

```diff
@@ -0,0 +1,154 @@
+use crate::utils::{
+    interface::{deposit, total_assets, total_supply},
+    setup::{defaults, deploy, get_wallet_balance, setup_nft, Deposit},
+};
+use fuels::types::Bits256;
+
+mod success {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn creates_vault() {
+        let (_deployer, admin, f_nft_id, nft_id) = deploy().await;
+        let (nft_1, nft_2) = setup_nft(&admin.wallet, &admin.nft, nft_id).await;
+        let (vault_sub_id, vault_admin, share_asset1, _share_asset2, share_supply) =
+            defaults(&admin.wallet, nft_1, nft_2, f_nft_id);
+
+        assert_eq!(get_wallet_balance(&admin.wallet, &share_asset1).await, 0);
+        assert_eq!(get_wallet_balance(&admin.wallet, &nft_1).await, 1);
+        assert_eq!(total_assets(&admin.f_nft).await, 0);
+        assert_eq!(total_supply(&admin.f_nft, share_asset1).await, None);
+
+        let response = deposit(&admin.f_nft, nft_1, vault_admin.clone(), vault_sub_id).await;
+
+        assert_eq!(
+            get_wallet_balance(&admin.wallet, &share_asset1).await,
+            share_supply
+        );
+        assert_eq!(get_wallet_balance(&admin.wallet, &nft_1).await, 0);
+        assert_eq!(total_assets(&admin.f_nft).await, 1);
+        assert_eq!(
+            total_supply(&admin.f_nft, share_asset1).await,
+            Some(share_supply)
+        );
+
+        let log = response.decode_logs_with_type::<Deposit>().unwrap();
+        let event = log.get(0).unwrap();
```

Comment:
```suggestion
        let event = log.first().unwrap();
```
Suggestion from my clippy